### ISSUE TITLE
IOError: [Errno22] Invalid Argument

### ISSUE THREAD

**Issue by ranadesameer** [frej/fast-export#103]
master: Exporting thorough delta revision 1884/3354 with 13/23/0 added/changed/removed files
Traceback (most recent call last):
  File "C:/Users/sranade/AppData/Local/Temp/fast-export/hg-fast-export.py", line 505, in <module>
    notes=options.notes,encoding=encoding,fn_encoding=fn_encoding))
  File "C:/Users/sranade/AppData/Local/Temp/fast-export/hg-fast-export.py", line 405, in hg2git
    sob,brmap,hgtags,encoding,fn_encoding)
  File "C:/Users/sranade/AppData/Local/Temp/fast-export/hg-fast-export.py", line 245, in export_commit
    export_file_contents(ctx,man,changed,hgtags,fn_encoding)
  File "C:/Users/sranade/AppData/Local/Temp/fast-export/hg-fast-export.py", line 141, in export_file_contents
    wr(d)
  File "C:/Users/sranade/AppData/Local/Temp/fast-export/hg-fast-export.py", line 37, in wr
    wr_no_nl(msg)
  File "C:/Users/sranade/AppData/Local/Temp/fast-export/hg-fast-export.py", line 34, in wr_no_nl
    sys.stdout.write(msg)
IOError: [Errno 22] Invalid argument

**Comment by frej** [comment]
This is not enough information to even try to begin to debug this. Please have a look to #79 (the part before "A workaround:") for an error report that gives me enough information to begin trying to figure out what is going wrong. Please provide a reference to the repo that triggers this error, or a small script that builds such a repo.

**Comment by ranadesameer** [comment]
I have a local repo in mercurial, I am trying to convert it to Git. 
![image](https://user-images.githubusercontent.com/28454908/30707614-e2d1e514-9eb0-11e7-95a8-9893f5b5ddd4.png)

After running this I got the above error.

**Comment by frej** [comment]
Sorry, without more information I can't help you.

Try to figure out what's particular about revision 1884, then create a
script that creates a hg-repo which crashes in the same way as the repo
you cannot share.

Also check the git fast-import log so this isn't just a case of
fast-import crashing and then triggering some strange io-error in
fast-export.

**Comment by ranadesameer** [comment]
Where can I see fast import logs?

**Comment by frej** [comment]
In .git/fast_import_crash_<pid>, desribed in the "Crash Reports" section
of the git fast-import man-page.

Be aware that fast-export runs asynchronously to fast-import, so it's
not unusual to see (if fast-exports feeds fast-import bad data) trace
printouts from fast export for commit up to no. <x+delta>, and then a
crash report from fast-export (typically a broken pipe). When you then
check the fast-import crash log you see that fast-import already crashed
on commit <x>.

**Comment by ranadesameer** [comment]
There was an error in my authors' file. After adding a dummy entry, I could resolve the error.

**Comment by jeff1089** [comment]
I got this error.  First I got the "FutureWarning" error and I got rid of that by uninstalling Python 3.7 and installing 3.6,  The same conversion on the same directory caused this error.  I looked at the Repo and there was no obvious issue with the check in in question.

I was able to resolve this eventually by noticing that a revision several versions before the last one processed had username "<>", which I think is what Mercurial uses instead of null.  Adding the -A parameter with "<>"="unknown" fixed the issue.
